### DISCUSSION TITLE
Start Server without specifying public ip

### DISCUSSION THREAD

**Discussion by Ahmed310** [jchristn/WatsonTcp#150] [Q&A]
It would be good to have an override function for server to start that takes port only, like most libraries provide.
Alternatively can you suggest an solution how I find that ip from application level (pragmatically) and pass it to that function.

I was exploring the TcpListener class it provides an overload that takes port only

`
using System.Threading.Tasks;

namespace System.Net.Sockets
{
public class TcpListener
{
[Obsolete("This method has been deprecated. Please use TcpListener(IPAddress localaddr, int port) instead. http://go.microsoft.com/fwlink/?linkid=14202")]
public TcpListener(int port);
public TcpListener(IPEndPoint localEP);
public TcpListener(IPAddress localaddr, int port);
public bool ExclusiveAddressUse { get; set; }
public EndPoint LocalEndpoint { get; }
public Socket Server { get; }
protected bool Active { get; }

    public static TcpListener Create(int port);
    public Socket AcceptSocket();
    public Task<Socket> AcceptSocketAsync();
    public TcpClient AcceptTcpClient();
    public Task<TcpClient> AcceptTcpClientAsync();
    public void AllowNatTraversal(bool allowed);
    public IAsyncResult BeginAcceptSocket(AsyncCallback callback, object state);
    public IAsyncResult BeginAcceptTcpClient(AsyncCallback callback, object state);
    public Socket EndAcceptSocket(IAsyncResult asyncResult);
    public TcpClient EndAcceptTcpClient(IAsyncResult asyncResult);
    public bool Pending();
    public void Start();
    public void Start(int backlog);
    public void Stop();
}
}
`

can we use it until its not removed

**Comment by jchristn** [comment]
It's not really needed on a server. I assume that's why they're deprecating it. Typically you would listen on 127.0.0.1, 0.0.0.0, *, or the actual IP address of an interface. Cheers, Joel

**Reply by jchristn** [reply]
If the goal is to just listen on a given port regardless on the IP address on which a connection was received, set the listener IP address to ```0.0.0.0``` or ```*```; that will listen on any available IP address.  It will require administrative privileges.
